core/math/vector2.h

```diff
@@ -81,56 +81,58 @@ struct [[nodiscard]] Vector2 {
 	Vector2 normalized() const;
 	bool is_normalized() const;
 
-	real_t length() const;
-	real_t length_squared() const;
+	_FORCE_INLINE_ real_t length() const { return Math::sqrt(length_squared()); }
+	_FORCE_INLINE_ real_t length_squared() const { return x * x + y * y; }
 	Vector2 limit_length(real_t p_len = 1.0) const;
 
-	Vector2 min(const Vector2 &p_vector2) const {
+	_FORCE_INLINE_ Vector2 min(const Vector2 &p_vector2) const {
 		return Vector2(MIN(x, p_vector2.x), MIN(y, p_vector2.y));
 	}
 
-	Vector2 minf(real_t p_scalar) const {
+	_FORCE_INLINE_ Vector2 minf(real_t p_scalar) const {
 		return Vector2(MIN(x, p_scalar), MIN(y, p_scalar));
 	}
 
-	Vector2 max(const Vector2 &p_vector2) const {
+	_FORCE_INLINE_ Vector2 max(const Vector2 &p_vector2) const {
 		return Vector2(MAX(x, p_vector2.x), MAX(y, p_vector2.y));
 	}
 
-	Vector2 maxf(real_t p_scalar) const {
+	_FORCE_INLINE_ Vector2 maxf(real_t p_scalar) const {
 		return Vector2(MAX(x, p_scalar), MAX(y, p_scalar));
 	}
 
-	real_t distance_to(const Vector2 &p_vector2) const;
-	real_t distance_squared_to(const Vector2 &p_vector2) const;
+	_FORCE_INLINE_ real_t distance_to(const Vector2 &p_vector2) const { return Math::sqrt(distance_squared_to(p_vector2)); }
+	_FORCE_INLINE_ real_t distance_squared_to(const Vector2 &p_vector2) const {
+		return (x - p_vector2.x) * (x - p_vector2.x) + (y - p_vector2.y) * (y - p_vector2.y);
```

Comment:
```suggestion
		real_t x2 = x - p_vector2.x;
		real_t y2 = y - p_vector2.y;
		return x2 * x2 + y2 * y2;
```

I think it's a good optimization for this function.

Reply:
How is it an optimization? Looks better, but best case scenario does the same thing.

Reply:
It generates less assembly instructions:
24 vs 31

Reply:
Not sure where you got that, with the optimization flag enabled it produces the same instructions:
https://godbolt.org/z/xx9ob8dGb

Reply:
Nevermind..

Reply:
For the record, I'm not against this change, but it'd be more for clarity's sake, not performance.